### PR TITLE
[zh-cn] update <script> element

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

updated content

### CODE REVIEW

File: files/zh-cn/web/html/element/script/index.md [hunk 1]
```diff
@@ -43,80 +49,184 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
 
+- `crossorigin`
+  - : 正常的 `script` 元素将最小的信息传递给 {{domxref('Window.error_event', 'window.onerror')}}，用于那些没有通过标准 {{Glossary("CORS")}} 检查的脚本。要允许对静态媒体使用独立域名的网站进行错误记录，请使用此属性。参见 [CORS 设置属性](/zh-CN/docs/Web/HTML/Attributes/crossorigin)，以获得对其有效参数的更多描述性解释。
 - `defer`
-  - : 这个布尔属性被设定用来通知浏览器该脚本将在文档完成解析后，触发 [`DOMContentLoaded`](/zh-CN/docs/Web/API/Window/DOMContentLoaded_event) 事件前执行。
-    有 `defer` 属性的脚本会阻止 `DOMContentLoaded` 事件，直到脚本被加载并且解析完成。
-    > **警告：** 如果缺少 `src` 属性（即内嵌脚本），该属性不应被使用，因为这种情况下它不起作用。`defer` 属性对模块脚本没有作用 —— 他们默认 defer。
+
+  - : 这个布尔属性的设置是为了向浏览器表明，该脚本是要在文档被解析后，但在触发 {{domxref("Document/DOMContentLoaded_event", "DOMContentLoaded")}} 事件之前执行的。
+
```
Comment (jasonren0403) [inline files/zh-cn/web/html/element/script/index.md:68]:
```suggestion

```

File: files/zh-cn/web/html/element/script/index.md [hunk 2]
```diff
@@ -43,80 +49,184 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
+  - : 正常的 `script` 元素将最小的信息传递给 {{domxref('Window.error_event', 'window.onerror')}}，用于那些没有通过标准 {{Glossary("CORS")}} 检查的脚本。要允许对静态媒体使用独立域名的网站进行错误记录，请使用此属性。参见 [CORS 设置属性](/zh-CN/docs/Web/HTML/Attributes/crossorigin)，以获得对其有效参数的更多描述性解释。
 - `defer`
-  - : 这个布尔属性被设定用来通知浏览器该脚本将在文档完成解析后，触发 [`DOMContentLoaded`](/zh-CN/docs/Web/API/Window/DOMContentLoaded_event) 事件前执行。
-    有 `defer` 属性的脚本会阻止 `DOMContentLoaded` 事件，直到脚本被加载并且解析完成。
-    > **警告：** 如果缺少 `src` 属性（即内嵌脚本），该属性不应被使用，因为这种情况下它不起作用。`defer` 属性对模块脚本没有作用 —— 他们默认 defer。
+
+  - : 这个布尔属性的设置是为了向浏览器表明，该脚本是要在文档被解析后，但在触发 {{domxref("Document/DOMContentLoaded_event", "DOMContentLoaded")}} 事件之前执行的。
+  
+    包含 `defer` 属性的脚本将阻塞 `DOMContentLoaded` 事件触发，直到脚本完成加载并执行。
+
```
Comment (jasonren0403) [inline files/zh-cn/web/html/element/script/index.md:70]:
```suggestion

```

File: files/zh-cn/web/html/element/script/index.md [hunk 3]
```diff
@@ -43,80 +49,182 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
+    - `no-referrer`：不会发送 {{HTTPHeader("Referer")}} 标头。
+    - `no-referrer-when-downgrade`（默认）：如果没有 {{Glossary("TLS")}}（{{Glossary("HTTPS")}}）协议，{{HTTPHeader("Referer")}} 标头将不会被发送到{{Glossary("origin","源")}}上。
+    - `origin`：发送的 referrer 将被限制在 referrer 页面的源：其[协议](/zh-CN/docs/Learn/Common_questions/Web_mechanics/What_is_a_URL)、{{Glossary("host","主机")}}和{{Glossary("port","端口")}}。
+    - `origin-when-cross-origin`：将会限制发送至其他源的 referrer 的协议、主机和端口号。在同源的导航上仍然包括路径。
+    - `same-origin`：在{{Glossary("Same-origin policy", "同源")}}内将发送 referrer，但是跨源请求不包含 referrer 信息。
+    - `strict-origin`：只在协议安全等级相同时（如 HTTPS→HTTPS）发送文档的源作为 referrer，目标安全性降低（如 HTTPS→HTTP）时不发送。
+    - `strict-origin-when-cross-origin`：在执行同源请求时，发送完整的 URL，但只在协议安全级别保持不变（如 HTTPS→HTTPS）时发送源，而在目标安全性降低（如 HTTPS→HTTP）时不发送标头。
+    - `unsafe-url`：referrer 将包含源*和*路径（但不包含[片段](/zh-CN/docs/Web/API/HTMLHyperlinkElementUtils/hash)、[密码](/zh-CN/docs/Web/API/HTMLHyperlinkElementUtils/password)和[用户名](/zh-CN/docs/Web/API/HTMLHyperlinkElementUtils/username)）。**这个值是不安全的**，因为它将 TLS 保护的资源的源和路径泄露给不安全的源。
+
```
Comment (jasonren0403) [inline files/zh-cn/web/html/element/script/index.md:106]:
```suggestion

```

File: files/zh-cn/web/html/element/script/index.md [hunk 4]
```diff
@@ -43,80 +49,182 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
-    关于如何引入特殊编程语言，请参见[这篇文章](/zh-CN/Add-ons/Code_snippets/Rosetta)。
-- `text`
-  - : 和 textContent 属性类似，本属性用于设置元素的文本内容。但和 textContent 不一样的是，本属性在节点插入到 DOM 之后，此属性被解析为可执行代码。
-
-### Deprecated attributes
+  - : 这个属性定义引用外部脚本的 URI，这可以用来代替直接在文档中嵌入脚本。
+- [**`type`**](/zh-CN/docs/Web/HTML/Element/script/type)
+
+  - : 该属性表示所代表的脚本类型。该属性的值可能为以下类型：
+
```
Comment (jasonren0403) [inline files/zh-cn/web/html/element/script/index.md:114]:
```suggestion

```

File: files/zh-cn/web/html/element/script/index.md [hunk 5]
```diff
@@ -43,80 +49,180 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
+  - : 表示在获取脚本或脚本获取资源时，要发送哪个 [referrer](/zh-CN/docs/Web/API/Document/referrer)：
+
+    - `no-referrer`：不会发送 {{HTTPHeader("Referer")}} 标头。
+    - `no-referrer-when-downgrade`（默认）：如果没有 {{Glossary("TLS")}}（{{Glossary("HTTPS")}}）协议，{{HTTPHeader("Referer")}} 标头将不会被发送到{{Glossary("origin","源")}}上。
+    - `origin`：发送的 referrer 将被限制在 referrer 页面的源：其[协议](/zh-CN/docs/Learn/Common_questions/Web_mechanics/What_is_a_URL)、{{Glossary("host","主机")}}和{{Glossary("port","端口")}}。
+    - `origin-when-cross-origin`：将会限制发送至其他源的 referrer 的协议、主机和端口号。在同源的导航上仍然包括路径。
+    - `same-origin`：在{{Glossary("Same-origin policy", "同源")}}内将发送 referrer，但是跨源请求不包含 referrer 信息。
+    - `strict-origin`：只在协议安全等级相同时（如 HTTPS→HTTPS）发送文档的源作为 referrer，目标安全性降低（如 HTTPS→HTTP）时不发送。
+    - `strict-origin-when-cross-origin`：在执行同源请求时，发送完整的 URL，但只在协议安全级别保持不变（如 HTTPS→HTTPS）时发送源，而在目标安全性降低（如 HTTPS→HTTP）时不发送标头。
+    - `unsafe-url`：referrer 将包含源*和*路径（但不包含[片段](/zh-CN/docs/Web/API/HTMLHyperlinkElementUtils/hash)、[密码](/zh-CN/docs/Web/API/HTMLHyperlinkElementUtils/password)和[用户名](/zh-CN/docs/Web/API/HTMLHyperlinkElementUtils/username)）。**这个值是不安全的**，因为它将 TLS 保护的资源的源和路径泄露给不安全的源。
```
Comment (jasonren0403) [inline files/zh-cn/web/html/element/script/index.md:105]:
```suggestion
    - `unsafe-url`：referrer 将包含源*和*路径（但不包含[片段](/zh-CN/docs/Web/API/HTMLAnchorElement/hash)、[密码](/zh-CN/docs/Web/API/HTMLAnchorElement/password)和[用户名](/zh-CN/docs/Web/API/HTMLAnchorElement/username)）。**这个值是不安全的**，因为它将 TLS 保护的资源的源和路径泄露给不安全的源。
```

File: files/zh-cn/web/html/element/script/index.md [hunk 6]
```diff
@@ -43,80 +49,180 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
+    - `no-referrer-when-downgrade`（默认）：如果没有 {{Glossary("TLS")}}（{{Glossary("HTTPS")}}）协议，{{HTTPHeader("Referer")}} 标头将不会被发送到{{Glossary("origin","源")}}上。
+    - `origin`：发送的 referrer 将被限制在 referrer 页面的源：其[协议](/zh-CN/docs/Learn/Common_questions/Web_mechanics/What_is_a_URL)、{{Glossary("host","主机")}}和{{Glossary("port","端口")}}。
+    - `origin-when-cross-origin`：将会限制发送至其他源的 referrer 的协议、主机和端口号。在同源的导航上仍然包括路径。
+    - `same-origin`：在{{Glossary("Same-origin policy", "同源")}}内将发送 referrer，但是跨源请求不包含 referrer 信息。
+    - `strict-origin`：只在协议安全等级相同时（如 HTTPS→HTTPS）发送文档的源作为 referrer，目标安全性降低（如 HTTPS→HTTP）时不发送。
+    - `strict-origin-when-cross-origin`：在执行同源请求时，发送完整的 URL，但只在协议安全级别保持不变（如 HTTPS→HTTPS）时发送源，而在目标安全性降低（如 HTTPS→HTTP）时不发送标头。
+    - `unsafe-url`：referrer 将包含源*和*路径（但不包含[片段](/zh-CN/docs/Web/API/HTMLAnchorElement/hash)、[密码](/zh-CN/docs/Web/API/HTMLAnchorElement/password)和[用户名](/zh-CN/docs/Web/API/HTMLAnchorElement/username)）。**这个值是不安全的**，因为它将 TLS 保护的资源的源和路径泄露给不安全的源。
+      > **注意**：空字符串（`""`）既是默认值，也是在不支持 `referrerpolicy` 的情况下的一个回退值。如果没有在 `<script>` 元素上明确指定 `referrerpolicy`，它将采用更高级别的 referrer 策略，即对整个文档或域设置的策略。如果没有更高级别的策略，空字符串将被视为等同于 `no-referrer-when-downgrade`。
```
Comment (jasonren0403) [inline files/zh-cn/web/html/element/script/index.md:106]:
```suggestion
      > **备注**：空字符串（`""`）既是默认值，也是在不支持 `referrerpolicy` 的情况下的一个回退值。如果没有在 `<script>` 元素上明确指定 `referrerpolicy`，它将采用更高级别的 referrer 策略，即对整个文档或域设置的策略。如果没有更高级别的策略，空字符串将被视为等同于 `no-referrer-when-downgrade`。
```

File: files/zh-cn/web/html/element/script/index.md [hunk 7]
```diff
@@ -43,80 +49,180 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
+    - `no-referrer-when-downgrade`（默认）：如果没有 {{Glossary("TLS")}}（{{Glossary("HTTPS")}}）协议，{{HTTPHeader("Referer")}} 标头将不会被发送到{{Glossary("origin","源")}}上。
+    - `origin`：发送的 referrer 将被限制在 referrer 页面的源：其[协议](/zh-CN/docs/Learn/Common_questions/Web_mechanics/What_is_a_URL)、{{Glossary("host","主机")}}和{{Glossary("port","端口")}}。
+    - `origin-when-cross-origin`：将会限制发送至其他源的 referrer 的协议、主机和端口号。在同源的导航上仍然包括路径。
+    - `same-origin`：在{{Glossary("Same-origin policy", "同源")}}内将发送 referrer，但是跨源请求不包含 referrer 信息。
+    - `strict-origin`：只在协议安全等级相同时（如 HTTPS→HTTPS）发送文档的源作为 referrer，目标安全性降低（如 HTTPS→HTTP）时不发送。
+    - `strict-origin-when-cross-origin`：在执行同源请求时，发送完整的 URL，但只在协议安全级别保持不变（如 HTTPS→HTTPS）时发送源，而在目标安全性降低（如 HTTPS→HTTP）时不发送标头。
+    - `unsafe-url`：referrer 将包含源*和*路径（但不包含[片段](/zh-CN/docs/Web/API/HTMLAnchorElement/hash)、[密码](/zh-CN/docs/Web/API/HTMLAnchorElement/password)和[用户名](/zh-CN/docs/Web/API/HTMLAnchorElement/username)）。**这个值是不安全的**，因为它将 TLS 保护的资源的源和路径泄露给不安全的源。
+      > **备注**：空字符串（`""`）既是默认值，也是在不支持 `referrerpolicy` 的情况下的一个回退值。如果没有在 `<script>` 元素上明确指定 `referrerpolicy`，它将采用更高级别的 referrer 策略，即对整个文档或域设置的策略。如果没有更高级别的策略，空字符串将被视为等同于 `no-referrer-when-downgrade`。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/script/index.md:106]:
```suggestion
      > **备注：** 空字符串（`""`）既是默认值，也是在不支持 `referrerpolicy` 的情况下的一个回退值。如果没有在 `<script>` 元素上明确指定 `referrerpolicy`，它将采用更高级别的 referrer 策略，即对整个文档或域设置的策略。如果没有更高级别的策略，空字符串将被视为等同于 `no-referrer-when-downgrade`。
```

**Review by yin1999 — COMMENTED** [review]
I don't like colons followed by spaces after "备注", but unfortunately that's the only format we can currently use (:

File: files/zh-cn/web/html/element/script/index.md [hunk 8]
```diff
@@ -43,80 +49,180 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
+  - : 这个属性明确指出，在获取脚本的过程中，某些操作应该被阻断。要阻断的操作必须是一个以空格分隔的列表，下面列出了阻断属性。
+    - `render`：屏幕上渲染内容的操作应该被阻断。
+
+### 废弃的属性
 
 - `charset` {{Deprecated_inline}}
-  - : 如果存在，值必须和“`utf-8`”不区分大小写的匹配。当然声明 `charset` 是没有必要的，因为页面文档必须使用 UTF-8，而 `script` 元素会从页面文档中继承这个属性。
-- `language` {{Deprecated_inline}}
-  - : 和 type 属性类似，这个属性定义脚本使用的语言。但是与 type 不同的是，这个属性的可能值从未被标准化过。请用`type`属性代替这个属性。
+  - : 如果存在，它的值必须是 ASCII 大小写不敏感的“`utf-8`”的匹配。没有必要指定 `charset` 属性，因为文档必须使用 UTF-8，而且 `script` 元素从文档中继承其字符编码。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/script/index.md:129]:
```suggestion
  - : 如果存在，它的值必须是 ASCII 大小写不敏感的“`utf-8`”的匹配。没有必要指定 `charset` 属性，因为文档必须使用 UTF-8，而且 `script` 元素从文档继承其字符编码。
```

File: files/zh-cn/web/html/element/script/index.md [hunk 9]
```diff
@@ -43,80 +49,180 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
+- `language` {{Deprecated_inline}} {{Non-standard_Inline}}
+  - : 和 `type` 属性类似，这个属性定义脚本使用的语言。但是与 `type` 不同的是，这个属性的可能值从未被标准化过。请用 `type` 属性代替这个属性。
+
+## 备注
+
+没有 [`async`](#async)、[`defer`](#defer) 或 `type="module"` 属性的脚本，以及没有 `type="module"` 属性的内联脚本，会在浏览器继续解析页面之前立即获取并执行。
+
+脚本应该以 `text/javascript` 的 MIME 类型提供，但浏览器比较宽容，只有在脚本以图像类型（`image/*`）、视频类型（`video/*`）、音频类型（`audio/*`）或 `text/csv` 提供时才会阻止它们。如果脚本受阻，将向该元素发送 {{domxref("Element/error_event", "error")}} 事件；否则，将发送 {{domxref("Element/load_event", "load")}} 事件。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/script/index.md:137]: mdn/content#27259

```suggestion
脚本应该以 `text/javascript` 的 MIME 类型提供，但浏览器比较宽容，只有在脚本以图像类型（`image/*`）、视频类型（`video/*`）、音频类型（`audio/*`）或 `text/csv` 提供时才会阻止它们。如果脚本受阻，将向该元素发送 {{domxref("HTMLElement/error_event", "error")}} 事件；否则，将发送 {{domxref("HTMLElement/load_event", "load")}} 事件。
```

File: files/zh-cn/web/html/element/script/index.md [hunk 10]
```diff
@@ -43,80 +49,180 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
+支持 `module` 作为 [`type`](#type) 属性的浏览器忽略任何具有 `nomodule` 属性的脚本。这种机制可以允许你在使用模块脚本时为不支持的浏览器提供 `nomodule` 标记的回落脚本。
 
-```plain
-<script type="module" src="main.mjs"></script>
+```html
+<script type="module" src="main.js"></script>
 <script nomodule src="fallback.js"></script>
 ```
 
+### 导入带有导入映射（importmap）的模块
```
Comment (yin1999) [inline files/zh-cn/web/html/element/script/index.md:166]: Let's update this with the following one

```suggestion
### 使用导入映射导入模块
```

File: files/zh-cn/web/html/element/script/index.md [hunk 11]
```diff
@@ -43,80 +49,180 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
+  - : 这个属性定义引用外部脚本的 URI，这可以用来代替直接在文档中嵌入脚本。
+- [**`type`**](/zh-CN/docs/Web/HTML/Element/script/type)
+
+  - : 该属性表示所代表的脚本类型。该属性的值可能为以下类型：
+    - **属性未设置（默认），一个空字符串，或一个 JavaScript MIME 类型**
+      - : 代表脚本为包含 JavaScript 代码的“传统的脚本”。如果脚本指的是 JavaScript 代码，我们鼓励作者省略这个属性，而不是指定一个 MIME 类型。所有的 JavaScript MIME 类型都列在 [IANA 的媒体类型规范](/zh-CN/docs/Web/HTTP/Basics_of_HTTP/MIME_types#textjavascript)中。
+    - `module`
+      - : 此值导致代码被视为 JavaScript 模块。其中的代码内容会延后处理。`charset` 和 `defer` 属性不会生效。对于使用 `module` 的更多信息，请参见 [JavaScript 模块](/zh-CN/docs/Web/JavaScript/Guide/Modules)指南。与传统代码不同的是，模块代码需要使用 CORS 协议来跨源获取。
+    - [`importmap`](/zh-CN/docs/Web/HTML/Element/script/type/importmap)
+      - : 此值代表元素体内包含导入映射表。导入映射表是一个 JSON 对象，开发者可以用它来控制浏览器在导入 [JavaScript 模块](/zh-CN/docs/Web/JavaScript/Guide/Modules#importing_modules_using_import_maps)时如何解析模块标识符。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/script/index.md:118]:
```suggestion
      - : 此值代表元素体内包含导入映射（importmap）表。导入映射表是一个 JSON 对象，开发者可以用它来控制浏览器在导入 [JavaScript 模块](/zh-CN/docs/Web/JavaScript/Guide/Modules#importing_modules_using_import_maps)时如何解析模块标识符。
```

File: files/zh-cn/web/html/element/script/index.md [hunk 12]
```diff
@@ -43,80 +49,180 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
+
+  - : 这个布尔属性的设置是为了向浏览器表明，该脚本是要在文档被解析后，但在触发 {{domxref("Document/DOMContentLoaded_event", "DOMContentLoaded")}} 事件之前执行的。
+    包含 `defer` 属性的脚本将阻塞 `DOMContentLoaded` 事件触发，直到脚本完成加载并执行。
+    > **警告：** 本属性不应在缺少 `src` 属性的情况下使用（也就是内联脚本的情况下），将不会生效。
+    >
+    > `defer` 属性对[模块脚本](/zh-CN/docs/Web/JavaScript/Guide/Modules)也不会生效——它们默认是 defer 的。
+
+    包含 `defer` 属性的脚本会按照它们出现在文档中的顺序执行。
+
+    这个属性能够消除**解析器阻塞的 JavaScript**，在这种情况下，浏览器必须在继续解析之前加载和执行脚本。`async` 在这种情况下也有类似的效果。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/script/index.md:75]: How about using "阻塞解析的"?

Comment (jasonren0403) [reply files/zh-cn/web/html/element/script/index.md:77]:
```suggestion
    这个属性能够消除**阻塞解析的 JavaScript**，在这种情况下，浏览器必须在继续解析之前加载和执行脚本。`async` 在这种情况下也有类似的效果。
```

File: files/zh-cn/web/html/element/script/index.md [hunk 13]
```diff
@@ -43,80 +49,182 @@ HTML `<script>` 元素用于嵌入或引用可执行脚本。这通常用作嵌
...
+      - : 获取该脚本的优先级比其他外部脚本的等级要低。
+    - `auto`
+      - : 默认值：自动确定获取该脚本的相对优先级。
+
 - `integrity`
-  - : 包含用户代理可用于验证已提取资源是否已无意外操作的内联元数据。参见 [Subresource Integrity](/zh-CN/docs/Web/Security/Subresource_Integrity)。
+  - : 包含用户代理可用于验证所获取到资源的完整性的内联元数据。参见[子资源完整性](/zh-CN/docs/Web/Security/Subresource_Integrity)。
 - `nomodule`
-  - : 这个布尔属性被设置来标明这个脚本在支持 [ES2015 modules](https://hacks.mozilla.org/2015/08/es6-in-depth-modules/) 的浏览器中不执行。 — 实际上，这可用于在不支持模块化 JavaScript 的旧浏览器中提供回退脚本。
+  - : 这个布尔属性被设置来标明这个脚本不应该在支持 [ES modules](/zh-CN/docs/Web/JavaScript/Guide/Modules) 的浏览器中执行。实际上，这可用于在不支持模块化 JavaScript 的旧浏览器中提供回退脚本。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/script/index.md:93]:
```suggestion
  - : 这个布尔属性被设置来标明这个脚本不应该在支持 [ES 模块](/zh-CN/docs/Web/JavaScript/Guide/Modules)的浏览器中执行。实际上，这可用于在不支持模块化 JavaScript 的旧浏览器中提供回退脚本。
```

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**